Marlin/src/inc/Warnings.cpp

```diff
@@ -797,3 +797,10 @@
 #if HAS_SHAPING && ANY(CORE_IS_XY, MARKFORGED_XY, MARKFORGED_YX)
   #warning "Input Shaping for CORE / MARKFORGED kinematic axes is still experimental."
 #endif
+
+/**
+ * BLTOUCH with PROBING_MARGIN
+ */
+#ifdef BLTOUCH_WITH_PROBING_MARGIN
+  #warning "A Probing margin is not required or rrecommended with a BLTOUCH"
+#endif
```

Comment:
This would not just apply to the BLTouch probe, but to all probes that are offset from the nozzle. Meanwhile, I consider it too alarmist to warn about the probing margin when there are valid reasons to keep it.

For example, a probe that is offset from the nozzle enough to limit the probing region will cause the ABL grid to be off-center; however, with a probing margin equal or greater than the probe offset, the grid will be centered. If that is the user's preference then we shouldn't nag them about it (nor require them to opt-out of the warning). If some users are complaining about the probing margin, the simple answer to them is to turn it off, letting them know that if they use ABL this may make the grid slightly more off-center. Of course this is not important for UBL.

Another valid reason to use a probing margin is because the edges of the bed tend to be more deflected than points closer to center, which can skew the mesh a little bit, and some probes (though not BLTouch) may push down enough on the bed before they trigger to cause significant deflection.

Maybe the long-term solution is to make the probing margin 0 by default, but for setups that strive to use the entire bed area, this could put the probe right at the edge of the bed. So in general at least a few mm is recommended for all probes, including the BLTouch.